### PR TITLE
Simplify base inheritance statement with Mojo::Base everywhere

### PR DESCRIPTION
**Pull request by okurz**
Wherever we use "Mojo::Base" as well as "base" we can simply combine the
both including some redundant "strict" and "warnings" statements which
are also implicitly included with Mojo::Base.

### CODE REVIEW

**Review by baierjan — APPROVED** [review]

**Review by Martchus — APPROVED** [review]

File: t/01-style.t [hunk 1]
```diff
@@ -13,4 +13,5 @@ qq{git grep -I -l 'This program is free software.*if not, see <http://www.gnu.or
 ok system(qq{git grep -I -l '[#/ ]*SPDX-License-Identifier ' ':!COPYING' ':!external/' ':!t/01-style.t'}) != 0,
   'SPDX-License-Identifier correctly terminated';
 is qx{git grep -I -L '^use Test::Most' t/**.t}, '', 'All tests use Test::Most';
+is qx{git grep -I -e '^use Mojo::Base' --and -e 'use base'}, '', 'No redundant Mojo::Base+base';
```
Comment (perlpunk) [inline t/01-style.t:16]: Is this command supposed to output anything in our current master?

**Review by perlpunk — CHANGES_REQUESTED** [review]
See my comment about git-grep

File: t/01-style.t (hunk 1 above)
Comment (okurz) [reply t/01-style.t:16]: Yes, doesn't it?

Comment (perlpunk) [reply t/01-style.t:16]: no:
```
% git describe
4.5.1528009330.e68ebe2b-6119-g6bb94d7ba
% git grep -I -e '^use Mojo::Base' --and -e 'use base'
% echo $?
1
% git --version
git version 2.26.2
```

Comment (okurz) [reply t/01-style.t:16]: Yes, I will cross-check to have a failing test first

Comment (okurz) [reply t/01-style.t]: Updated with a better match rule that lists all the files changed in this PR before the patch is applied. Seems I confused commands when I included the previous one in here without crosschecking after applying if it's still the correct one. Thanks for diligent checking.

**Review by perlpunk — APPROVED** [review]

**Pull request merged**